structuralcodes/sections/_generic.py

```diff
@@ -775,26 +775,92 @@ def calculate_moment_curvature(
         return res
 
     def calculate_nm_interaction_domain(
-        self, theta: float = 0, num_axial: int = 100
+        self,
+        theta: float = 0,
+        n1: int = 1,
+        n2: int = 2,
+        n3: int = 15,
+        n4: int = 10,
+        n5: int = 3,
+        n6: int = 4,
+        n: t.Optional[int] = None,
+        type1: t.Literal['linear', 'geometric', 'quadratic'] = 'linear',
+        type2: t.Literal['linear', 'geometric', 'quadratic'] = 'linear',
+        type3: t.Literal['linear', 'geometric', 'quadratic'] = 'geometric',
+        type4: t.Literal['linear', 'geometric', 'quadratic'] = 'linear',
+        type5: t.Literal['linear', 'geometric', 'quadratic'] = 'linear',
+        type6: t.Literal['linear', 'geometric', 'quadratic'] = 'linear',
     ) -> s_res.NMInteractionDomain:
         """Calculate the NM interaction domain.
 
         Arguments:
-        theta (float, default = 0): inclination of n.a. respect to y axis
-        n_axial (int, default = 100): number of discretization for axial load
+        theta (float): inclination of n.a. respect to y axis
+            (Optional, default = 0)
+        n1: (int) number of strain profiles in field 1 (Optional, default = 1)
+        n2: (int) number of strain profiles in field 2 (Optional, default = 2)
+        n3: (int) number of strain profiles in field 3 (Optional, default = 15)
+        n4: (int) number of strain profiles in field 4 (Optional, default = 10)
+        n5: (int) number of strain profiles in field 5 (Optional, default = 3)
+        n6: (int) number of strain profiles in field 6 (Optional, default = 4)
+        n: (int) total number of strain profiles (Optional, default = None)
+            If specified n and n1..n6 the total number of n may be different.
+        type1: (literal) type of spacing for field 1
+            'linear' for a linear spacing 'geometric' for a geometric spacing
+            (Optional default = 'linear')
+        type2: (literal) type of spacing for field 2
+            'linear' for a linear spacing 'geometric' for a geometric spacing
+            (Optional default = 'linear')
+        type3: (literal) type of spacing for field 3
+            'linear' for a linear spacing 'geometric' for a geometric spacing
+            (Optional default = 'geometric')
+        type4: (literal) type of spacing for field 4
+            'linear' for a linear spacing 'geometric' for a geometric spacing
+            (Optional default = 'linear')
+        type5: (literal) type of spacing for field 5
+            'linear' for a linear spacing 'geometric' for a geometric spacing
+            (Optional default = 'linear')
+        type6: (literal) type of spacing for field 6
+            'linear' for a linear spacing 'geometric' for a geometric spacing
+            (Optional default = 'linear')
```

Comment:
Good that you also included these descriptions in the docstring.

Please slightly rewrite to comply with our format. Write e.g. `num_1 (int): Number of ... .` Such that you have the variable name, parenthesis with type, colon, description starting with uppercase letter and ending with period 😃